content/blog/2024/kubewarden-1-10-release.md

```diff
@@ -0,0 +1,129 @@
+
+For example, if a user has the apparmor policy deployed 5 times (with different
+names, settings, etc), now only one instance of PolicyEvaluator will be
+allocated for all of those 5 deployments.
+
+In addition, we rely now on Wasmtime's pre-initialization of Wasm modules. This
+allows us to keep a deduplicated list of all the policies in use, in their
+pre-initialize form. This list is shared between all policy evaluators in the
+policy-server.
+
+#### Protection against policy mem leaks, and clean slate per policy
+
+Before, there were long-running policy evaluators, ready to process policy requests.
+
+Now, policy evaluation mode happens on-demand: we receive a request, a fresh
+policy evaluator is created, used just for evaluation that request, and
+discarded.
+
+This is possible as we now leverage Wasmtime's pre-initialization, hence
+creation time of policy evaluators in greatly reduced.
+
+Using this on-demand approach to policy evaluators brings 2 main advantages:
+
+- Impact of policies leaking memory is now virtualy non-existent.
+- Each evaluation starts with a clean slate, preventing bugs caused by policies
+  leaving unclean state between evaluations.
+
+#### Codebase simplification and increased test coverage
+
+Also, thanks to a big refactor, together with additional unit and integration
+tests, the policy-server codebase has moved away from using a worker thread
+pool for policy evaluation and into an async approach, removing codebase complexity.
+
+Stay tunned for more performance improvements!
+
+## WASI policies written in Go can now be context aware
+
+Thanks to recent improvements in the official Go compiler, we have added
+context-aware capabilities via WASI on to the kubewarden Go SDK, hence it is
+now possible to write context-aware Go WASI policies.
+
+As an example, have a look at the
+[go-wasi-context-aware-test-policy](https://github.com/kubewarden/go-wasi-context-aware-test-policy).
+
+See the WASI updated docs [here](TODO).
+
+## Controller
+
+The controller now reconciles divergences bewtween `ClusterAdmissionPolicies`
+and `AdmissionPolicies` and their associated webhook objects, mitigating
+possible attacks which delete or modify policy webhooks, changing or stopping
+policies from working. (these attacks where already mitigated by RBAC rights).
+
+## Audit Scanner
+
+Thanks to the community, the Audit Scanner can now be run with an in-memory store
+for policy reporting. By default, it keeps using Etcd and the PolicyReports CRDs.
+This paves the way to implementing more performant storage solutions for policy
+reporting, such a relational database backend.
+See `--set auditScanner.store` in the kubewarden-controller chart for
+configuring it.
+
+## Kwctl
+
+When doing `kwctl inspect` we now hide signatures by default for brevity. One
+can still obtain them using `kwctl inspect --show-signatures`. In addition,
+`kwctl inspect -o yaml` will produce sane YAML.
```

Comment:
Was it insane previously? :-). I'd take out the word unless it previously wasn't correct.
```suggestion
`kwctl inspect -o yaml` provides alternative output.
```

Reply:
Maybe sane is a big word, but before, it wasn't correct YAML, it's a bug fix.

Reply:
reworded to "correct" with https://github.com/kubewarden/kubewarden.io/pull/210/commits/ab70dd8186f8ca7c10ca16d6052a65fa5fd0389b.